src/dali/transaction_resolver.py

```diff
@@ -232,69 +233,73 @@ def resolve_transactions(
     transaction: AbstractTransaction
 
     try:
-        for transaction in transactions:
-            if not isinstance(transaction, AbstractTransaction):
-                raise RP2RuntimeError(f"Internal error: Parameter 'transaction' is not a subclass of AbstractTransaction. {transaction}")
+        with ProgressBar(max_value=len(transactions)) as progress_bar:
+            for transaction in transactions:
+                if not isinstance(transaction, AbstractTransaction):
+                    raise RP2RuntimeError(f"Internal error: Parameter 'transaction' is not a subclass of AbstractTransaction. {transaction}")
+
+                if transaction.fiat_ticker not in {None, global_configuration[Keyword.NATIVE_FIAT.value]}:
+                    # Foreign exchanges may have transactions denominated in non-native fiat
+                    transaction = _convert_fiat_fields_to_native_fiat(transaction, global_configuration)
+
+                if is_unknown(transaction.unique_id):
+                    # Cannot resolve further if unique_id is not known
+                    if read_spot_price_from_web:
+                        transaction = _update_spot_price_from_web(transaction, global_configuration)
+                    LOGGER.debug("Unresolvable transaction (no %s): %s", Keyword.UNIQUE_ID.value, str(transaction))
+                    resolved_transactions.append(transaction)
+                else:
+                    transaction_list: List[AbstractTransaction]
+                    transaction_list = unique_id_2_transactions.setdefault(AssetAndUniqueId(transaction.asset, transaction.unique_id), [])
+                    transaction_list.append(transaction)
+                    unique_id_2_transactions[AssetAndUniqueId(transaction.asset, transaction.unique_id)] = transaction_list
+
+        with ProgressBar(max_value=len(unique_id_2_transactions.values())) as progress_bar:
```

Comment:
Would it be better to have just one progress bar? In other words have only one `with ProgressBar` at line 236, but initialize it this way:
```
        with ProgressBar(max_value=len(transactions) + len(unique_id_2_transactions)) as progress_bar:
```
And place both loops in the scope of the `with`. 

Also, would it be possible to display just the percentage, without the absolute number (which may or may not make sense to the user)?

Reply:
Yes, that'd be better. The issue is that `unique_id_2_transactions` isn't populated until the first for loop, but I think it might be possible to set `max_value=2*len(transactions)` and then update `max_value` once the first for loop is done (so might jump from 50% to 100% if the latter is empty).

And yes, the output is totally configurable but I might not be able to finalize that today, so it might be a few days before this is ready to merge.

Reply:
The two progress bars are a bit confusing. I think updating the max_value midway sounds like a reasonable solution. We don't need to get too fancy.